### ISSUE TITLE
Add exceptions for the "no-callback-in-promise" rule

### ISSUE THREAD

**Issue by EvgenyOrekhov** [xjamundx/eslint-plugin-promise#43]
Sometimes a callback isn't a "nodeback", as with middleware:
```js
function middleware(req, res, next) {
    return promise(req)
        .then(function (result) {
            res.send(result);
            return next();
        })
        .catch(next);
}
```
It is impossible to use noedify in this case, so it should be possible to add "next" to exceptions.

**Comment by xjamundx** [comment]
One thing we could do is allow callbacks and nexts, but enforce that they're used properly. (Like if you `next()` in a `then()` it has to always have 0 arguments and it also needs to be found in a `catch`. Might be hard to get right though. I still think you can `nodeify` this code though if you wanted too and use it for middleware without too much trouble.

**Comment by EvgenyOrekhov** [comment]
@xjamundx In restify, for example, `next` can accept a route name or `false` as the first argument, in those cases nodeify wouldn't work.

**Comment by xjamundx** [comment]
@EvgenyOrekhov thanks for the clarification, I'm fine making the list of callback names configurable

**Comment by EvgenyOrekhov** [comment]
@xjamundx I'd prefer the "opt-out" policy, i. e. when you specify a list of what not to include:
```json
"no-callback-in-promise": [
    "error",
    {
        "exceptions": ["next"]
    }
]
```

**Comment by xjamundx** [comment]
I guess for me the benefit of opt-in is because someone may have a different convention for callbacks like they might call them `done` and so with this option they could include both, but I'm sort of fine either way :)

**Comment by EvgenyOrekhov** [comment]
@xjamundx The more linters and rules we use, the better our programs will be, that's why I think a rule should warn on all cases by default and a programmer should think twice before making an exception to a rule.

**Comment by xjamundx** [comment]
sure the opt-in list would be optional, but default it would include `next`
:)

**Comment by jshemas** [comment]
Any updates on this?

**Comment by xjamundx** [comment]
I probably can't do it myself, but if you submit a PR I can probably get it in there!

**Comment by jshemas** [comment]
Done! Let me know what you think.

**Comment by xjamundx** [comment]
Will publish these today, thanks!

**Comment by macklinu** [comment]
This was resolved in #68 and shipped in v3.6.0. Closing.

**Comment by GuyMograbi** [comment]
works for me. 

just added this to my `.eslintrc`

```json
"promise/no-callback-in-promise": ["error", { "exceptions": ["next"] }],
```
